### PR TITLE
Added max_objects, object_index and total_count variable for pagination

### PR DESCRIPTION
**Pull request by siva-portworx**
- Added max_objects, object_index and total_count variable for pagination
- Added stats service with stats get API.

### CODE REVIEW

File: pkg/apis/v1/api.proto [hunk 1]
```diff
@@ -526,6 +526,7 @@ message EnumerateOptions {
     // Filter to use for cluster name of objects. Any object that contains the
     // filter will be returned
     string cluster_name_filter = 6;
+    int64 page_index = 7;
```
Comment (disrani-px) [inline pkg/apis/v1/api.proto:529]:
```suggestion
    uint64 page_index = 7;
```

File: pkg/apis/v1/api.proto [hunk 2]
```diff
@@ -1133,6 +1134,7 @@ message BackupEnumerateResponse {
     repeated BackupObject backups = 1;
     // continuation token
     string continuation_token = 2;
+    int64 total_count = 3;
```
Comment (disrani-px) [inline pkg/apis/v1/api.proto:1137]:
```suggestion
    uint64 total_count = 3;
```

**Review by disrani-px — APPROVED** [review]

File: pkg/apis/v1/api.proto (hunk 1 above)
Comment (siva-portworx) [reply pkg/apis/v1/api.proto]: Fixed

File: pkg/apis/v1/api.proto (hunk 2 above)
Comment (siva-portworx) [reply pkg/apis/v1/api.proto]: Fixed

File: pkg/apis/v1/api.proto [hunk 3]
```diff
@@ -515,9 +515,7 @@ message EnumerateOptions {
     // label selectors for the object for filtering
     map<string, string> labels = 1;
     // max objects to fetch
-    int64 page_size = 2;
-    // continuation token
-    string continuation_token = 3;
+    uint64 max_objects = 2;
     // Time range for which to return objects
     TimeRange time_range = 4;
```
Comment (disrani-px) [inline pkg/apis/v1/api.proto:520]: Update to 3 and other values below too

File: pkg/apis/v1/api.proto [hunk 4]
```diff
@@ -1049,6 +1048,26 @@ message BackupLocationValidateRequest {
...
+        option (google.api.http) = {
+            get : "/v1/backup/{org_id}"
+        };
+    }
+}
+
+message StatsGetRequest { string org_id = 1; }
+
+message StatsGetResponse {
+    repeated Stats backup_stats = 1;
```
Comment (disrani-px) [inline pkg/apis/v1/api.proto:1062]:
```suggestion
    repeated Stats stats = 1;
```

File: pkg/apis/v1/api.proto [hunk 5]
```diff
@@ -1049,6 +1048,26 @@ message BackupLocationValidateRequest {
...
 
+service Stats {
+    rpc Get(StatsGetRequest) returns (StatsGetResponse) {
+        option (google.api.http) = {
+            get : "/v1/backup/{org_id}"
+        };
+    }
+}
+
+message StatsGetRequest { string org_id = 1; }
```
Comment (disrani-px) [inline pkg/apis/v1/api.proto:1059]: Use Inspect to keep it consistent with other APIs
```suggestion
message MetricsInspecttRequest { string org_id = 1; }
```

And let's call it Metrics

**Review by disrani-px — APPROVED** [review]

File: pkg/apis/v1/api.proto (hunk 3 above)
Comment (siva-portworx) [reply pkg/apis/v1/api.proto:520]: Fixed.

File: pkg/apis/v1/api.proto (hunk 4 above)
Comment (siva-portworx) [reply pkg/apis/v1/api.proto:1062]: Fixed.

File: pkg/apis/v1/api.proto (hunk 5 above)
Comment (siva-portworx) [reply pkg/apis/v1/api.proto]: Fixed

**Pull request merged**